src/lsm/segmented_array.zig

```diff
@@ -1255,6 +1255,21 @@ fn FuzzContextType(
                 );
                 try testing.expectEqual(iterator_end.next(), null);
             }
+
+            {
+                // 0 is not symmetric with maxInt, because `array.search` doesn't take direction
+                // into account.
```

Comment:
:exploding_head: Surprising!

I double-checked the call sites to see if they were buggy:

`ForestTableIterator.next()` is fine, it only ever iterates ascending.

`ManifestLevel.iterator_start()` is more confusing. I _think_ the code is correct thanks to `iterator_start_boundary()`, but this might be accidental.

It has a comment which notes this behavior:

```
            // Ascending:  Find the first table where table.key_max ≤ iterator.key_min.
            // Descending: Find the first table where table.key_max ≤ iterator.key_max.
            const target = level.keys.search(switch (direction) {
                .ascending => key_min,
                .descending => key_max,
            });
```

...And I think in the comment, the inequalities are backwards?